mikrotik/resource_bgp_instance.go

```diff
@@ -115,10 +115,6 @@ func resourceBgpInstanceRead(d *schema.ResourceData, m interface{}) error {
 
 	bgpInstance, err := c.FindBgpInstance(d.Id())
 
-	// TODO: Ignoring this error can cause all resources to think they
-	// need to be created. We should more appropriately handle this. The
-	// error where the record is not found is not actually an error and
-	// needs to be disambiguated from real failures
 	if err != nil {
```

Comment:
You need to identify the `NotFound` error here.

Reply:
Oh, whoops, I had an extra `!` that I carried over from `bgp_instance_test.go`, but I'm dumb and forgot to get rid of it.